src/storage/src/source/postgres.rs

```diff
@@ -286,6 +286,8 @@ pub enum DefiniteError {
     MissingColumn,
     #[error("failed to parse COPY protocol")]
     InvalidCopyInput,
+    #[error("The replication timeline id has changed. Expected {0} but got {1}")]
+    InvalidTimelineId(u32, u32),
```

Comment:
Rust supports more structured errors where you can name the fields––given that it isn't obvious which is which to the reader, I'd suggest using that e.g.

```rust
Enum {
  expected: u32,
  actual: u32,
}
```